Neos.ContentRepository.Core/Classes/Projection/ContentGraph/ContentSubgraphInterface.php

```diff
@@ -141,12 +151,15 @@ public function findBackReferences(NodeAggregateId $nodeAggregateId, Filter\Find
     public function countBackReferences(NodeAggregateId $nodeAggregateId, Filter\CountBackReferencesFilter $filter): int;
 
     /**
-     * Find a single node underneath $startingNodeAggregateId that matches the specified $path
+     * Find a single node underneath
+     * - the given root node type name if defined in the path
+     * - the given starting node aggregate ID else
+     * that matches the specified $path
      *
      * NOTE: This operation is most likely to be deprecated since the concept of node paths is not really used in the core, and it has some logical issues
      * @return Node|null the node that matches the given $path, or NULL if no node on that path is accessible
      */
-    public function findNodeByPath(NodePath $path, NodeAggregateId $startingNodeAggregateId): ?Node;
+    public function findNodeByPath(NodePath $path, ?NodeAggregateId $startingNodeAggregateId): ?Node;
```

Comment:
im not sure i like this conditional behaviour cluttered into one method ...

Would it really make the api easier to use?

what do you think of this alternative, is it too complex?
```php
$node = $path->isAbsolute()
   ? $subgraph->findNodeByAbsolutePath($path)
   : $subgraph->findNodeByPath($path, $startingNodeAggregateId)
```

Reply:
Also do not like the second argument that can be null depending on the first one that much. For me it would be fine to always require a nodeAggregateId.

The alternative would be to have a distinct AbsoluteRootPath value object wich is a bit overdone.

Reply:
yes, the more I think about it the more I prefer separate VOs and separate methods.

Reply:
Say hello to AbsoluteNodePath and ContentSubgraphInterface::findNodeByAbsolutePath